backend/inspirehep/search/facets.py

```diff
@@ -300,44 +306,51 @@ def results=[:];
 
 
 def records_jobs():
+    order = count(start=1)
     return {
         "filters": {**current_app.config["JOBS_FILTERS"]},
         "aggs": {
-            **jobs_field_of_interest_aggregation(order=1),
-            **jobs_rank_aggregation(order=2),
-            **jobs_region_aggregation(order=3),
+            **jobs_field_of_interest_aggregation(order=next(order)),
+            **jobs_rank_aggregation(order=next(order)),
+            **jobs_region_aggregation(order=next(order)),
         },
     }
 
 
 def records_conferences():
+    order = count(start=1)
     return {
         "filters": {**current_app.config["CONFERENCES_FILTERS"]},
-        "aggs": {**conf_subject_aggregation(order=1)},
+        "aggs": {**conf_subject_aggregation(order=next(order))},
     }
 
 
 def records_hep_cataloger():
     records = records_hep()
-    records["aggs"].update({**hep_collection_aggregation(order=8)})
+    order = count(start=8)
+    records["aggs"].update({**hep_collection_aggregation(order=next(order))})
     return records
 
 
 def hep_author_publications_cataloger():
     author_recid = request.values.get("author", "", type=str).split("_")[0]
     records = hep_author_publications()
+    order = count(start=7)
     records["aggs"].update(
         {
-            **hep_subject_aggregation(order=7),
-            **hep_arxiv_categories_aggregation(order=8),
-            **hep_self_author_names_aggregation(order=9, author_recid=author_recid),
-            **hep_collection_aggregation(order=10),
+            **hep_subject_aggregation(order=next(order)),
+            **hep_arxiv_categories_aggregation(order=next(order)),
+            **hep_self_author_names_aggregation(
+                order=next(order), author_recid=author_recid
+            ),
+            **hep_collection_aggregation(order=next(order)),
         }
     )
     return records
 
 
 def records_jobs_cataloger():
     records = records_jobs()
-    records["aggs"].update({**jobs_status_aggregation(order=4)})
+    order = count(start=4)
```

Comment:
for the cases of cataloger facets, I want to have a function that gives me the last order of the copied aggregations but then of course, I hit nested aggregations and  would have needed to look into nested dictionaries to retrieve the meta (as I don't know at which level it is for each one), or I could add a useless meta just for this at the level I expect it to be. Do you think it is worth it?

Reply:
you can also add `order` as a kwarg for all facet functions instead, like
```
def records_jobs_cataloger(order=count(start=1))
```
In the caller, you can then inject the parent caller if needed.

Reply:
yeah, that's a good idea, thanks. But the count can't be the default argument to the function, I will need to default it to None and add a check 
`    if order is None:
        order = count(start=1)` because counter is mutable